WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift

```diff
@@ -882,6 +882,34 @@ extension ShippingLabelFormViewModel {
     }
 }
 
+// MARK: - Shipping Notice dismiss state handling
+extension ShippingLabelFormViewModel {
+    func setEUShippingNoticeDismissState(isDismissed: Bool,
+                                         onCompletion: @escaping (Bool) -> Void) {
+        let action = AppSettingsAction.setEUShippingNoticeDismissState(isDismissed: isDismissed) { result in
+            switch result {
+            case .success:
+                onCompletion(true)
+            case .failure:
+                onCompletion(false)
+            }
+        }
+        stores.dispatch(action)
+    }
+
+    func shouldDisplayEUShippingNotice(onCompletion: @escaping (Bool) -> Void) {
+        let action = AppSettingsAction.loadEUShippingNoticeDismissState { result in
+            switch result {
+            case .success(let dismissed):
+                onCompletion(!dismissed && self.isEUShippingNotificationEnabled)
+            case .failure:
+                onCompletion(false)
+            }
+        }
+        stores.dispatch(action)
+    }
```

Comment:
This works just fine, but as an alternative: What about checking `isEUShippingNotificationEnabled` first and returning early (skipping the action entirely) if it's `false`? Just an idea!

Reply:
Way better in terms of performance! Fixed in https://github.com/woocommerce/woocommerce-ios/pull/9695/commits/ca5d0cb9cedd5e363ca11b77140217f137583bd6.

Thanks!